app/src/common/shared/com/igalia/wolvic/speech/MKSpeechRecognizer.java

```diff
@@ -54,10 +65,24 @@ public void start(@NonNull Settings settings, @NonNull Callback callback) {
         if (StringUtils.isEmpty(key)) {
             key = DEBUG_API_KEY;
         }
+
+        // locale set by the app's configuration
+        Locale defaultLocale = mContext.getResources().getConfiguration().getLocales().get(0);
+        if (defaultLocale == null) {
+            // default locale for this JVM
+            defaultLocale = Locale.getDefault();
+        }
+
         if (settings.locale.equals(LocaleUtils.DEFAULT_LANGUAGE_ID)) {
             settings.locale = LocaleUtils.getClosestLanguageForLocale(
-                    Locale.getDefault(), mSupportedLanguages, LocaleUtils.FALLBACK_LANGUAGE_TAG);
+                    defaultLocale, mSupportedLocales, LocaleUtils.FALLBACK_LANGUAGE_TAG);
+        } else if (mSupportedLanguages.contains(settings.locale)) {
+            // TODO use the device's location to find the country code
+            Locale userLocale = new Locale(settings.locale, defaultLocale.getCountry());
+            settings.locale = LocaleUtils.getClosestLanguageForLocale(
+                    userLocale, mSupportedLocales, LocaleUtils.FALLBACK_LANGUAGE_TAG);
```

Comment:
What about (pseudocode)
```
if (local == DEFAULT)
   locale = defaultLocale
else if (supported.contains(locale)
   locale=  new Locale()

settings.locale = LocaleUtils...(locale, ....)
```

Reply:
I ended up with:

```java
if (supported.contains(locale)
   locale = new Locale()
else
   locale=  defaultLocale

settings.locale = LocaleUtils...(locale, ....)
```

This ensures that the `locale` variable is always initialised.